ccsf/cs132a_ruby/assn_02/arr2cols.rb

```diff
@@ -97,38 +97,31 @@
 ]
 
 # create dimensional array
-all_arrays = [ escargot_player_data, employees, artists ]
+all_arrays = [
+  [escargot_player_data, "Table 1: Escargot Players Data"], 
+  [employees, "Table 2: Employees"], 
+  [artists, "Table 3: Artist Addresses and Income"]
+]
 
-puts "Table 1: Escargot Players Data"
-# Iterate thru 1st element in all_arrays which contains Escargot Players Data.
-# Print each row in 14-space columns using String#%.
-all_arrays[0].each do |arr|
-# added "|" to printf so rows a little easier to read.
-# left justified with %-<value>s for readability
-  arr.each {|item| printf("%-14s" , item) }
-    puts
-end
+all_arrays.each do |array|
+  puts array.last()
 
-puts
+  maxVal = []
+  array.first().each do |row|
+    row.each_with_index do |col, i|
+      maxVal[i] = [maxVal.at(i) || 0, col.to_s.length].max
```

Comment:
nevermind. I reverse engineered it and figured it out. Nice job!

row.each_with_index loop is iterating over each value in the array and resolving it's length. Then it populates maxVal array with that length. 

row :["Jim", "bullfrog", 99]
lengths of each value = [3,8,2]

On 1st iteration, maxVal is an empty array. So [maxVal.at(index) returns nil which resolves to false. So the or || statement triggers a return of 0. This auto creates an array of [0, item.to_s.length].  
Then it resolves the highest value via .max method. 

In the case of value 'Jim' , array contains [0,3] and .max method returns 3.

Then maxVal is populated with value, 3 which is set at 0 index position. maxVal = [3]. 

This is repeated on all remaining items in the array, with each new max value set to a different index in maxVal. Until each value in the row has been evaluated.

So after the loop executes on the 1st row :["Jim", "bullfrog", 99], maxVal = [3,8,2].

Now that maxVal is populated with values, the 1st or || condition  'maxVal.at(index)' no longer returns nil. But returns the max length value that resolved in the previously iterated row.

 So after the loop executes on the 2nd row: ["Mack the Knife", "caterpillar", 12]
who's values has lengths: [14,11,2]
and max method returns the largest number.

maxVal is updated to = [14,11,2].

this is how all columns in the escargot_player_data 2 dimensional array is evaluated to find out the length of the longest value. And it's used to find out the optimal column width so data columns are  printed to screen and no data is cut off.